mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/KeyValueBasedPersistenceProvider.java

```diff
@@ -371,7 +372,14 @@ public List<IMantisJobMetadata> loadAllJobs() throws IOException {
                     continue;
                 }
                 for (MantisWorkerMetadataWritable workerMeta : workersByJobId.get(jobId)) {
-                    jobMeta.addWorkerMedata(workerMeta.getStageNum(), workerMeta, null);
+                    Preconditions.checkState(
+                        jobMeta.addWorkerMedata(workerMeta.getStageNum(), workerMeta, null),
+                        "JobID=%s stage=%d workerIdx=%d has existing worker, existing=%s, new=%s",
+                        workerMeta.getJobId(),
+                        workerMeta.getStageNum(),
+                        workerMeta.getWorkerIndex(),
+                        jobMeta.getWorkerByIndex(workerMeta.getStageNum(), workerMeta.getWorkerIndex()).getWorkerId(),
+                        workerMeta.getWorkerId());
                 }
                 jobMetas.add(DataFormatAdapter.convertMantisJobWriteableToMantisJobMetadata(jobMeta, eventPublisher));
```

Comment:
Another thing worth logging (or even exception) is non-contiguous utilization of `workerIdx`. Mantis doesn't handle gaps very well and so proactively logging it when adding job to jobMetas would be a good idea.

Reply:
This is a separate problem, and I will tackle this separately. But, for the time being, the script I have catches this issue.